src/main/java/org/jabref/gui/exporter/ExportToClipboardAction.java

```diff
@@ -159,22 +161,17 @@ private void setContentToClipboard(ExportResult result) {
     }
 
     private String readFileToString(Path tmp) throws IOException {
-        try (BufferedReader reader = Files.newBufferedReader(tmp, panel.getBibDatabaseContext()
-                                                                       .getMetaData()
-                                                                       .getEncoding()
-                                                                       .orElse(preferences.getGeneralPreferences().getDefaultEncoding()))) {
+        Charset defaultEncoding = Objects.requireNonNull(preferences.getGeneralPreferences().getDefaultEncoding());
+        try (BufferedReader reader = Files.newBufferedReader(tmp, stateManager.getActiveDatabase()
+                                                                              .map(db -> db.getMetaData()
+                                                                                           .getEncoding()
+                                                                                           .orElse(defaultEncoding))
+                                                                              .orElse(defaultEncoding))) {
             return reader.lines().collect(Collectors.joining(OS.NEWLINE));
         }
     }
 
-    private static class ExportResult {
-
-        final String content;
-        final FileType fileType;
+    private record ExportResult(String content, FileType fileType) {
```

Comment:
Please wait with this new Syntax elements. We are still at jdk 16

Reply:
Records are already a stable language feature since jdk 16.
https://openjdk.java.net/jeps/395